hedera-node/hedera-app-spi/src/test/java/com/hedera/node/app/spi/state/WrappedWritableKVStateTest.java

```diff
@@ -57,4 +60,69 @@ void commitGoesToDelegateNotBackingStore() {
         assertThat(delegate.get(B_KEY)).isEqualTo(BLACKBERRY); // Has the new value
         assertThat(delegate.get(E_KEY)).isEqualTo(ELDERBERRY); // Has the new value
     }
+
+    @Nested
+    @DisplayName("size")
+    final class SizeTest {
+        @Test
+        @DisplayName("Adding a key that is not in the backing store impacts the size")
+        void putNew() {
+            assertThat(state.readKeys()).isEmpty();
+            assertThat(state.modifiedKeys()).isEmpty();
+
+            // Before inserting, the size of backing store should be 2 and modifications are none
+            assertEquals(2, state.size());
+            assertEquals(2, delegate.size());
+            assertEquals(0, delegate.modifiedKeys().size());
+
+            state.put(C_KEY, CHERRY);
+
+            // After doing a put, the size is increased as modifications are considered.
+            // But the modifiedKeys or size of delegate doesn't change until commit
+            assertEquals(3, state.size());
+            assertEquals(2, delegate.size());
+            assertEquals(0, delegate.modifiedKeys().size());
+
+            // Commit should not cause the size of backing store to be increased by 1.
+            // Instead, modifications on delegate have increased.
+            // Since modifications are increased, size of delegate also increases.
+            state.commit();
+            Mockito.verify(state, Mockito.times(1)).putIntoDataSource(anyString(), anyString());
+            Mockito.verify(state, Mockito.times(1)).putIntoDataSource(C_KEY, CHERRY);
```

Comment:
Is the first line needed? Seems like the second line would check the same thing, but more specifically

Reply:
yeah, not needed. I will remove